src/compiler/checker.ts

```diff
@@ -22983,7 +22983,11 @@ namespace ts {
                     return getContextuallyTypedParameterType(declaration);
                 case SyntaxKind.BindingElement:
                     return getContextualTypeForBindingElement(declaration);
-                // By default, do nothing and return undefined - only parameters and binding elements have context implied by a parent
+                case SyntaxKind.PropertyDeclaration:
+                    if (hasSyntacticModifier(declaration, ModifierFlags.Static)) {
+                        return getContextualTypeForStaticPropertyDeclaration(declaration);
+                    }

```

Comment:
Do we care about this case (non-static), or is it already covered somehow?

```ts
interface Ctor {
  prototype: {
    opts: {
      len: 'foo' | 'bar'
    }
  }
}
let myVar: Ctor = class { opts = { len: 'foo' } };
```

Reply:
It's definitely not handled, but `prototype` lookup is, like.... weird but OK I guess? Testing
```ts
class Ctor {
    opts: {
      len: 'foo' | 'bar'
    }
}
let myVar: typeof Ctor = class { opts = { len: 'foo' } };
```
would also be pertinent if we find this important. We'd definitely need to explicitly handle it - I _think_ something like
`getTypeOfPropertyOfContextualType(getTypeOfPropertyOfContextualType(parentType, "prototype" as __String), getSymbolOfNode(declaration).escapedName)` might work; but I don't think we do any other direct prototype lookups in contextual typing as of yet, so it'd definitely be unique at least. Plus, there's also the constructor signature case to think about - `new () => InstanceType` - where any `.prototype` linkage between the static side and the instance is only implied, at best.

In an argument position, ie
```ts
declare function expectsClassWithLiteralMember<T extends string | number | symbol, C extends {prototype: { member: T }}>(member: T, cls: C): C;
expectsClassWithLiteralMember("ok", class X<T> { member = this.method(); method() { return null as T; } });
```
it has potential to play in odd ways with inference, since the `prototype` is usually the uninferred class instance.

It might be best to just not open that can of worms.

Reply:
I talked to Ryan about it, and he thinks that using `prototype` is weird enough to leave it as is, and address it on demand if that happens.